### DISCUSSION TITLE
Results output to both XML and console

### DISCUSSION THREAD

**Discussion by DaStoned** [cgreen-devs/cgreen#290] [Ideas]
I'm a bit surprised to find that I can only choose one output for test results - e.g. text or XML but not both. 

In a CI system I would generate the XML and feed it into the JUnit plugin for pretty graphs. However, I most certainly still need the text output from the same unit test run, as this is the most useful input to the developer so they can quickly understand what failed and fix it. This is how other UT frameworks, e.g. GTest, do it. 

Am I missing something or is this by design?

**Comment by thoni56** [comment]
Thanks for your input!

It is currently not possible to use multiple reporters in the same run as the underlying Cgreen library does not support this. This feature has not been disregarded, just not discussed before.

The simple workaround would of course be to run the tests twice in your CI. If the tests are not taking prohibitively long time to execute (they shouldn't ;-) that would be a fairly simple fix. Especially if you are using the `cgreen-runner`, then it would just be a matter of duplicating the invocation command with different options.

**Reply by DaStoned** [reply]
Yeah, thanks - that's the workaround I used. It accounts for some rather clumsy logic, as I need to tell Jenkins to ignore the result code from one of them (otherwise it would immediately fail the entire step and not execute the other run). It also assumes my tests are deterministic and idempotent - which happens to be the case so far, conveniently :)
